### ISSUE TITLE
`.pyi` files are not discovered when running on a folder

### ISSUE THREAD

**Issue by DetachHead** [pylint-dev/pylint#9097]
### Bug description

```python
# asdf/foo.py
foo = 1

# asdf/bar.pyi
foo = 1
```

### Command used

```shell
pylint asdf
```

### Pylint output

```
************* Module foo
asdf\foo.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
asdf\foo.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
```

### Expected behavior

```
************* Module foo
asdf\foo.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
asdf\foo.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
asdf\bar.pyi:1:0: C0104: Disallowed name "foo" (disallowed-name)
asdf\bar.pyi:1:0: C0104: Disallowed name "foo" (disallowed-name)
```

### Pylint version

```shell
pylint 3.0.0a7
astroid 3.0.0
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
```

### OS / Environment

windows 10
